File: dlr_ros_utils/src/tf_listener.cpp
#include <ros/ros.h>
#include <tf2_ros/transform_listener.h>
#include <geometry_msgs/TransformStamped.h>

int main(int argc, char** argv)
{
    ros::init(argc, argv, "my_tf2_listener");

    ros::NodeHandle nh("~");

    tf2_ros::Buffer tf_buffer;
    tf2_ros::TransformListener tfListener(tf_buffer);

    if (argc != 3)
    {
        ROS_ERROR_STREAM("Provided " << argc-1 << " arguments. Need parent and child frame");
        return -1;
    };

    std::string parent_frame, child_frame;
    parent_frame = argv[1];
    child_frame = argv[2];

    ros::Rate rate(1.0);

    while (ros::ok())
    {
        try
        {
            geometry_msgs::TransformStamped transform_stamped = 
                        tf_buffer.lookupTransform(parent_frame, child_frame, ros::Time(0));
            ROS_INFO("TF received.");
            ROS_INFO_STREAM(transform_stamped);
        }
        catch (tf2::TransformException &ex) 
        {
            ROS_WARN("%s",ex.what());
        }

        rate.sleep();
    }
    return 0;
};

File: dlr_ros_utils/scripts/tf_listener.py
#!/usr/bin/env python3

import rospy
import tf2_ros
import geometry_msgs.msg
import sys

if __name__ == '__main__':
    rospy.init_node('my_tf2_listener')

    tf_buffer = tf2_ros.Buffer()
    tf_listener = tf2_ros.TransformListener(tf_buffer)

    if len(sys.argv) != 3:
        rospy.logerr("Provided " + str(len(sys.argv)-1) + " arguments. Need parent and child frame.")
        exit()

    parent_frame = sys.argv[1]
    child_frame = sys.argv[2]

    rate = rospy.Rate(1.0)

    while not rospy.is_shutdown():
        try:
            transform_stamped = tf_buffer.lookup_transform(parent_frame, child_frame, rospy.Time())
            rospy.loginfo("TF received.")
            rospy.loginfo(transform_stamped)
        except (tf2_ros.LookupException, tf2_ros.ConnectivityException, tf2_ros.ExtrapolationException):
            rospy.logwarn("No TF from " + parent_frame + " to " + child_frame + " found.")
        
        rate.sleep()
